### ISSUE TITLE
Major 'inconsistent text' problem when handling empty cells

### ISSUE THREAD

**Issue by martinreynaert** [proycon/foliapy#28]
Hi,

I have 568 fine FoLiA xml files, obtained first by running tei2folia, tokenized next by Ucto. I need to add lemmatization and POS by way of spacy2folia. 

This unexpectedly fails on 164, or almost 29% of these, unfortunately.

Far as I can see their stderr messages all report the same issue. I give you the core of the smallest file's error report:

folia.main.ParseError: FoLiA exception in handling of <div> @ line 61 (in parent <text> @ parent line 60) : [InconsistentText] Text for <Division at 140369425161904 id=Collingwood-A-1935-PastMasters_eng_Collingwood_Philosophical_Texts_2nd_Release_The_Historical_Imagination-V1.text.div.1.div.1 set=https://raw.githubusercontent.com/proycon/folia/master/setdefinitions/tei2folia/divisions.foliaset.ttl class=book>, is inconsistent: EXPECTED (deep text after normalization) *****>
Frontmatter Titlepage {ii} {1} THE HISTORICAL IMAGINATION An Inaugural Lecture DELIVERED BEFORE THE UNIVERSITY OF OXFORD ON 28 OCTOBER 1935 By R. G. COLLINGWOOD WAYNFLETE PROFESSOR OF METAPHYSICAL PHILOSOPHY OXFORD AT THE CLARENDON PRESS 1935 {2} Copyright page OXFORD UNIVERSITY PRESS AMEN HOUSE, E.C. 4 London | Edinburgh | Glasgow New York | Toronto | Melbourne Capetown | Bombay | Calcutta Madras | Shanghai HUMPHREY MILFORD PUBLISHER TO THE UNIVERSITY PRINTED IN GREAT BRITAIN {3}
****> BUT FOUND (strict text after normalization) ****>
Frontmatter Titlepage {ii} {1} THE HISTORICAL IMAGINATION An Inaugural Lecture DELIVERED BEFORE THE UNIVERSITY OF OXFORD ON 28 OCTOBER 1935 By R. G. COLLINGWOOD WAYNFLETE PROFESSOR OF METAPHYSICAL PHILOSOPHY OXFORD AT THE CLARENDON PRESS 1935 {2} Copyright page OXFORD UNIVERSITY PRESS AMEN HOUSE, E.C. 4 London | Edinburgh | Glasgow New York | Toronto | Melbourne Capetown | Bombay | Calcutta Madras | Shanghai | HUMPHREY MILFORD PUBLISHER TO THE UNIVERSITY PRINTED IN GREAT BRITAIN {3}
******* DEVIATION POINT:  Shanghai <*HERE*>| HUMPHREY
(also checked against older rules prior to FoLiA v2.4.1)

I could get you the tokenized FoLiA version of this one.

I hope this can expediently be solved!

Thank you, as ever.

**Comment by proycon** [comment]
This relates to https://github.com/proycon/foliatools/issues/41

**Comment by martinreynaert** [comment]
Hi proycon,

Thank you for working on this problem!

Alas, the issue is not completely solved. Of the 164 files that failed with the previous release, still 44 fail with this new one.

The error messages are much the same. Would it be possible to further look into this, please!

I will get you two of the still failing files via my usual route. Both validate with my folialint. Foliavalidator does find deviation points in both.

Hope those help!

**Comment by proycon** [comment]
In these two files that still fail, we see an empty row with empty cells:

Error (excerpt):

```
Expected (deep text after normalization):  of bees. 387 |
Found (strict text):  of bees. | | 387 
******* DEVIATION POINT:  of bees. <*HERE*>| | 387 |
```

Original text:

```
359 | 288, n. 366, 292-3 | Hermann Müller on sexual differences of bees.
  |   |  
387 | 308 | Sounds produced by moths.
```

FoLiA XML:

```xml
                  <w xml:id="Darwin-A-1877-PastMasters_eng_The_Descent_of_Man_and_Selection_in_Relation_to_Sex_PartI-V1.text.div.1.div.1.div.5.table.3.row.9.cell.3.s.1.w.7" class="WORD" space="no">
                    <t>bees</t>
                  </w>
                  <w xml:id="Darwin-A-1877-PastMasters_eng_The_Descent_of_Man_and_Selection_in_Relation_to_Sex_PartI-V1.text.div.1.div.1.div.5.table.3.row.9.cell.3.s.1.w.8" class="PUNCTUATION">
                    <t>.</t>
                  </w>
                </s>
              </cell>
            </row>
            <row xml:id="Darwin-A-1877-PastMasters_eng_The_Descent_of_Man_and_Selection_in_Relation_to_Sex_PartI-V1.text.div.1.div.1.div.5.table.3.row.10">
              <cell xml:id="Darwin-A-1877-PastMasters_eng_The_Descent_of_Man_and_Selection_in_Relation_to_Sex_PartI-V1.text.div.1.div.1.div.5.table.3.row.10.cell.1"/>
              <cell xml:id="Darwin-A-1877-PastMasters_eng_The_Descent_of_Man_and_Selection_in_Relation_to_Sex_PartI-V1.text.div.1.div.1.div.5.table.3.row.10.cell.2"/>
              <cell xml:id="Darwin-A-1877-PastMasters_eng_The_Descent_of_Man_and_Selection_in_Relation_to_Sex_PartI-V1.text.div.1.div.1.div.5.table.3.row.10.cell.3"/>
            </row>
            <row xml:id="Darwin-A-1877-PastMasters_eng_The_Descent_of_Man_and_Selection_in_Relation_to_Sex_PartI-V1.text.div.1.div.1.div.5.table.3.row.11">
              <cell xml:id="Darwin-A-1877-PastMasters_eng_The_Descent_of_Man_and_Selection_in_Relation_to_Sex_PartI-V1.text.div.1.div.1.div.5.table.3.row.11.cell.1">
                <t> 387 </t>
                <s xml:id="Darwin-A-1877-PastMasters_eng_The_Descent_of_Man_and_Selection_in_Relation_to_Sex_PartI-V1.text.div.1.div.1.div.5.table.3.row.11.cell.1.s.1">
                  <t>387</t>
                  <w xml:id="Darwin-A-1877-PastMasters_eng_The_Descent_of_Man_and_Selection_in_Relation_to_Sex_PartI-V1.text.div.1.div.1.div.5.table.3.row.11.cell.1.s.1.w.1" class="NUMBER">
                    <t>387</t>
                  </w>
                </s>
              </cell>
```

The reconstructed indeed misses two textdelimiters for cell.

**Comment by proycon** [comment]
@martinreynaert I implemented a fix again. Can you try the development version first? (`pip install -U git+https://github.com/proycon/foliapy`), just to ensure they all pass now prior to release.

**Comment by martinreynaert** [comment]
This has helped, thanks, proycon!

Six of 44 files now fail on what seem to be right-to-left characters, though. Only foliavalidator sees these problems, neither tei2folia, uct or folialint do.

Can you please fix this too?

**Comment by proycon** [comment]
Fixed in v2.5.11
